### PR TITLE
Foundry 0.9 Compatibility and some refactoring

### PR DESCRIPTION
**Pull request by alessiocali**
Hi! Looks like 0.9 broke the module again 😃 So here's a fix. 

The first commit takes care of removing all references to the Entity API in favor of the new document API, and bumps the compatibility. That's the actual fix.

The second commit is something I took the liberty to add: I noticed the click handling function was duplicated and inconsistent, and also a bit hard to follow. So I moved into a single function and cleaned up the flow a bit to improve readability (up to my tastes, to be fair).

### CODE REVIEW

**Comment by jegasus** [comment]
Holy crap! You've done all the work here! This is incredible, thank you! Sorry it's taken me a while to respond. I can't look into this right this instant, but I'll definitely look at it this weekend!

**Comment by alessiocali** [comment]
No worries! Glad to be of help 😁

**Comment by jegasus** [comment]
This is awesome, my dude! I'm about to merge it to the main branch. Would it be okay if I added you to the "authors" section in the module manifest? If so, let me know which of the following details you'd like me to add:

  {
    "name": "Name of the author",
    "url": "https://website.com/of/the/author",
    "email": "email@example.com",
    "discord": "discordID#0001",
    "ko-fi": "kofiName",
    "patreon": "patreonName",
    "reddit": "u/RedditUsername",
    "twitter": "@TwitterHandle",
  }

**Comment by alessiocali** [comment]
Sure ! You can add the following if you like 🙂

```
{
"name": "Alessio Giuseppe Cali'",
"url": "https://alessiogcali.com/"
}
```
